todo/src/main/java/org/openapitools/controller/UserApiController.java

```diff
@@ -0,0 +1,78 @@
+import org.openapitools.model.entity.User;
+import org.openapitools.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.stereotype.Controller;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.context.annotation.SessionScope;
+import org.springframework.web.context.request.NativeWebRequest;
+
+@Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2023-04-02T20:47:24.972943100+02:00[Europe/Warsaw]")
+@Controller
+@RequestMapping("${openapi.todo.base-path:}")
+public class UserApiController implements UserApi {
+
+	private final NativeWebRequest request;
+	private final UserService userService;
+
+	@Autowired
+	public UserApiController(NativeWebRequest request, UserService userService) {
+		this.request = request;
+		this.userService = userService;
+	}
+
+	@Override
+	public Optional<NativeWebRequest> getRequest() {
+		return Optional.ofNullable(request);
+	}
+
+	@Override
+	public ResponseEntity<Void> login(UserLoginRequest userLoginRequest) throws ApiException {
+		User user = userService.loginUser(userLoginRequest);
+		MultiValueMap<String, String> headers = new LinkedMultiValueMap<>();
+		headers.add("userId", String.valueOf(user.getId()));
+		return new ResponseEntity<>(headers, HttpStatus.OK);
+
+	}
+
+	@Override
+	public ResponseEntity<Void> register(UserRegisterRequest userRegisterRequest) throws ApiException {
+		User user = userService.registerUser(userRegisterRequest);
+		MultiValueMap<String, String> headers = new LinkedMultiValueMap<>();
+		headers.add("userId", String.valueOf(user.getId()));
+		return new ResponseEntity<>(headers, HttpStatus.OK);
+	}
+
+	@Override
+	public ResponseEntity<String> getUserInfo() throws ApiException{
+		Principal token = request.getUserPrincipal();
+		String userName;
+		if(token instanceof OAuth2AuthenticationToken){
+			OAuth2AuthenticationToken defaultToken = (OAuth2AuthenticationToken) request.getUserPrincipal();
+			 userName = defaultToken.getPrincipal().getAttributes().get("name").toString();
+			return new ResponseEntity<> (userName, HttpStatus.OK);
+		} else {
+			throw new ApiException(HttpStatus.FORBIDDEN.value());
```

Comment:
You can implement verification if user has logged in using OAuth or not using roles. 
Please assign a role "OAuth_User" to users that have logged in via Google. When you have this you can change security config for UserInfo endpoint to `.requestMatchers("/user/info").hasRole("OAUTH_USER")`

Reply:
Done